libs/estdlib/src/supervisor.erl

```diff
@@ -364,6 +553,132 @@ try_start(#child{start = {M, F, Args}} = Record) ->
             {error, {{'EXIT', Error}, Record}}
     end.
 
+terminate_one_for_all_restart(Children) ->
+    %% Always shut down last child first
+    do_terminate_one_for_all_restart(lists:reverse(Children), [], []).
+
+do_terminate_one_for_all_restart([], NewChildren, []) ->
+    %% Do not reverse the list here, it was reversed before being accumulated
+    %% and is now in correct startup order.
+    {ok, NewChildren};
+do_terminate_one_for_all_restart([], NewChildren, WaitExit) ->
+    %% Do not reverse the list here, it was reversed before being accumulated
+    %% and is now in correct startup order.
+    {ok, NewChildren, WaitExit};
+do_terminate_one_for_all_restart([Child | Children], NewChildren, WaitExit) ->
+    case Child of
+        #child{restart = {terminating, temporary, From}} = Child when is_pid(From) ->
+            do_terminate(Child),
+            case verify_shutdown(Child) of
```

Comment:
I would queue all children for verification by default and avoid the 100ms wait here. Then I would loop until all exit message are received. I wonder about building a list of children to timeout kill instead of using the timers, but the timers can make a cleaner loop:

```erlang
terminate_loop([]) ->
    ok;
terminate_loop(Children) ->
    receive
        {'EXIT', Pid, _Reason} ->
            terminate_loop(lists:keydelete(Pid, #child.pid, Children));
        {ensure_killed, Pid} ->
            exit(Pid, kill),
            terminate_loop(Children)
    end.
```